qa/rpc-tests/wallet_listreceived.py

```diff
@@ -435,9 +442,101 @@ def test_received_sapling(self, height):
         assert_equal(r[1]['blockindex'], -1) # not yet mined
         assert 'blocktime' in r[1]
 
+    def test_received_orchard(self, height):
+        self.generate_and_sync(height+1)
+        taddr = self.nodes[1].getnewaddress()
+        acct1 = self.nodes[1].z_getnewaccount()['account']
+        acct2 = self.nodes[1].z_getnewaccount()['account']
+
+        addrResO = self.nodes[1].z_getaddressforaccount(acct1, ['orchard'])
+        assert_equal(addrResO['pools'], ['orchard'])
+        uao = addrResO['unifiedaddress']
+
+        addrResSO = self.nodes[1].z_getaddressforaccount(acct2, ['sapling', 'orchard'])
+        assert_equal(addrResSO['pools'], ['sapling', 'orchard'])
+        uaso = addrResSO['unifiedaddress']
+
+        self.nodes[0].sendtoaddress(taddr, 4.0)
+        self.generate_and_sync(height+2)
+
+        opid = self.nodes[1].z_sendmany(taddr, [
+            {'address': uao, 'amount': 1, 'memo': my_memo},
+            {'address': uaso, 'amount': 2},
+        ])
+        txid0 = wait_and_assert_operationid_status(self.nodes[1], opid)
+        self.sync_all()
+
+        # Decrypted transaction details should be correct, even though
+        # the transaction is still just in the mempool
+        pt = self.nodes[1].z_viewtransaction(txid0)
+
+        assert_equal(pt['txid'], txid0)
+        assert_equal(len(pt['spends']), 0)
+        assert_equal(len(pt['outputs']), 2)
+
+        # Outputs are not returned in a defined order but the amounts are deterministic
+        outputs = sorted(pt['outputs'], key=lambda x: x['valueZat'])
+        assert_equal(outputs[0]['type'], 'orchard')
+        assert_equal(outputs[0]['address'], uao)
+        assert_equal(outputs[0]['value'], Decimal('1'))
+        assert_equal(outputs[0]['valueZat'], 100000000)
+        assert_equal(outputs[0]['output'], 0)
+        assert_equal(outputs[0]['outgoing'], False)
+        assert_equal(outputs[0]['memo'], my_memo)
+        assert_equal(outputs[0]['memoStr'], my_memo_str)
+
+        assert_equal(outputs[1]['type'], 'orchard')
+        assert_equal(outputs[1]['address'], uaso)
+        assert_equal(outputs[1]['value'], Decimal('2'))
+        assert_equal(outputs[1]['valueZat'], 200000000)
+        assert_equal(outputs[1]['output'], 1)
+        assert_equal(outputs[1]['outgoing'], False)
+        assert_equal(outputs[1]['memo'], no_memo)
+        assert 'memoStr' not in outputs[1]
+
+        self.generate_and_sync(height+3)
+
+        opid = self.nodes[1].z_sendmany(uao, [
+            {'address': uaso, 'amount': Decimal('0.4')}
+        ])
```

Comment:
Add a recipient address from a different node, so we can confirm we see `outgoing: true`.